service/src/main/java/bio/terra/workspace/app/controller/ReferencedGcpResourceController.java

```diff
@@ -920,6 +923,60 @@ public ResponseEntity<ApiCloneReferencedGitRepoResourceResult> cloneGitRepoRefer
     return new ResponseEntity<>(result, HttpStatus.OK);
   }
 
+  // - Terra workspace referenced resource - //
+  @Override
+  public ResponseEntity<ApiTerraWorkspaceResource> createTerraWorkspaceReference(
```

Comment:
We really need another controller for non-cloud-based references. This will be the third, I think after TDR snapshot and git repo. I made [PF-1749](https://broadworkbench.atlassian.net/browse/PF-1749) to track.